lib/Agrammon/DB/Dataset.pm6

```diff
@@ -286,6 +296,25 @@ class Agrammon::DB::Dataset does Agrammon::DB {
         return self;
     }
 
+    method upload-data($content) {
+        my $fh = IO::String.new($content);
+        my $csv = Text::CSV.new;
+        my $i = 0;
+        while (my @row = $csv.getline($fh)) {
+            my ($var-name, $value) = @row;
+            next unless $var-name;
+            # skip comments
+            next if $var-name ~~ /^\#/;
+            self.store-input($var-name, $value);
+            $i++;
+        }
+        CATCH {
+            # DB failure
+            die X::Agrammon::DB::Dataset::UploadFailed.new(:dataset-name($!name));
+        }
```

Comment:
I guess it should also be a CSV format failure; is it worth differentiating those?

Reply:
I added three separate exceptions (CSV, Database, Unkown).